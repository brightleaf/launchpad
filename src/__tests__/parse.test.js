const path = require('path')
const parser = require('../parse')
describe('The YML Parser Module', () => {
  it('should parse a yml file', () => {
    // do stuff
    const result = parser(path.resolve(__dirname, './config.yml'))

    expect(result.service).toBe('launchpad')
    expect(result).toMatchSnapshot()
  })
})
