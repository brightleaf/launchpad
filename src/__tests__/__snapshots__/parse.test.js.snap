// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The YML Parser Module should parse a yml file 1`] = `
Object {
  "functions": Object {
    "firstMethod": Object {
      "description": "Something awesome.",
      "events": Array [
        Object {
          "http": Object {
            "cors": true,
            "method": "GET",
            "path": "/sample",
          },
        },
      ],
      "handler": "sample/index.first",
    },
    "secondMethod": Object {
      "description": "Something else that is awesome.",
      "events": Array [
        Object {
          "http": Object {
            "cors": true,
            "method": "GET",
            "path": "/sample",
          },
        },
      ],
      "handler": "sample/index.second",
    },
  },
  "provider": Object {
    "name": "aws",
    "runtime": "nodejs12.x",
  },
  "service": "launchpad",
}
`;
